### PR TITLE
Assorted stunbaton and stunglove fixes

### PR DESCRIPTION
**Pull request by Garen7**
## About The Pull Request

This is a bunch of bug fixes related to stunbatons and stungloves(the stunglove code is a copy paste of stunbatons so it fits together in the PR)

## Why It's Good For The Game

The only change that wasn't a bugfix was swapping the default cell in the batons and I did that because medium high cells already spawn in the armory so swapping the cell out roundstart was kinda tedious, and I put an excelsior cell in the excelsior baton because its from the same faction.

Also the excelsior medium cell can be printed from the same disk as the excelsior baton and since its a direct upgrade that's cheaper to print than the default medium cell it wouldn't make sense for the user to not immediately print an excelsior cell and swap out the batteries.

Also I didn't add a new sprite even though the .dmi is edited, I just copy pasted the excelsior baton sprite for the nocell excelsior baton sprite. I put up a request for a new sprite [here](https://discord.com/channels/255035529085583360/617459024472047646/770370644717731850) though its not vital.

## Changelog
:cl:
tweak: The normal baton now spawns with a medium high cell and the excelsior baton now spawns with an excelsior cell
fix: Fixed the excelsior baton glowing orange instead of cyan when on
fix: Fixed the excelsior baton not using its turned on inhand sprite when turned on
fix: Fixed batons sometimes keeping their pulsing quality after being turned off
fix: Fixed batons and stungloves not turning off when out of power
fix: Fixed batons and stungloves having an extra use even when out of power
fix: Fixed baton and stunglove sprites not updating when the cell is removed or added
fix: Fixed stungloves not showing their current cell charge when examined
fix: Fixed stungloves action button not updating
admin: Added logging for stungloves in the attack logs
/:cl:

### CODE REVIEW

**Review by mattroks101 — APPROVED** [review]

File: code/modules/clothing/gloves/stungloves.dm [hunk 1]
```diff
@@ -88,9 +92,10 @@
 	else
 		L.visible_message(SPAN_DANGER("[L] has been punched with [src] by [user]!"))
 	playsound(loc, 'sound/weapons/Egloves.ogg', 50, 1, -1)
-	L.stun_effect_act(stun, agony, user.targeted_organ, src)
-	msg_admin_attack("[key_name(user)] stunned [key_name(L)] with the [src].")
-	deductcharge(hitcost)
+
+	if(deductcharge(hitcost))
+		L.stun_effect_act(stun, agony, user.targeted_organ, src)
+		msg_admin_attack("[key_name(user)] stunned [key_name(L)] with the [src].")
```
Comment (TheShown911) [inline code/modules/clothing/gloves/stungloves.dm:99]: Not sure admins should be messaged upon a stun, attack logs should be enough.

Comment (Garen7) [reply code/modules/clothing/gloves/stungloves.dm:99]: I didn't add this, it was already there before hand, did you want me to remove it still?

That being said, I don't know if attack logs register for help intent and the admins should know whether the stun actually went through probably

File: code/game/objects/items/weapons/stunbaton.dm [hunk 2]
```diff
@@ -18,14 +18,15 @@
 	var/status = FALSE		//whether the thing is on or not
 	var/hitcost = 100
 	var/obj/item/weapon/cell/cell
-	var/spawn_cell = TRUE
+	var/obj/item/weapon/cell/starting_cell = /obj/item/weapon/cell/medium/high
```
Comment (Evankhell561) [inline code/game/objects/items/weapons/stunbaton.dm:21]: this new variable is unnecessary. you can work based on "cell" and suitable "cell".

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/stunbaton.dm [hunk 3]
```diff
@@ -17,16 +17,15 @@
 	var/agonyforce = 40
 	var/status = FALSE		//whether the thing is on or not
 	var/hitcost = 100
-	var/obj/item/weapon/cell/cell
-	var/obj/item/weapon/cell/starting_cell = /obj/item/weapon/cell/medium/high
+	var/obj/item/weapon/cell/cell = /obj/item/weapon/cell/medium/high
```
Comment (Evankhell561) [inline code/game/objects/items/weapons/stunbaton.dm:20]: uhm, nice work but not what i expected and this is out of step with the rest of the code. would you mind rolling back this commit? and I apologize for the inconvenience.

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/stunbaton.dm (hunk 3 above)
Comment (Garen7) [reply code/game/objects/items/weapons/stunbaton.dm:20]: uhh sure, what did you want out of curiosity? like
```suggestion
	var/obj/item/weapon/cell/cell = new /obj/item/weapon/cell/medium/high()
```
It works but I can't find anywhere else in the code that does something like that out of a proc so I wasn't sure if people were avoiding it for some reason

File: code/modules/clothing/gloves/stungloves.dm (hunk 1 above)
Comment (Garen7) [reply code/modules/clothing/gloves/stungloves.dm:99]: Just checked apparently it doesn't message admins or make a log entry, either way im looking into it now

Comment (Garen7) [reply code/modules/clothing/gloves/stungloves.dm:99]: nevermind the msg_admin_attack thing is a preference thing, and this IS needed for that, an important issue on the side though is still that stun gloves arent logged in the attack logs

Comment (Garen7) [reply code/modules/clothing/gloves/stungloves.dm:104]: fixed

**Comment by Garen7** [comment]
image
added 2 new fixes from the newest commit to the changelog

**Review by Hopekz — APPROVED** [review]

**Review by TorinoFermic — APPROVED** [review]

**Pull request merged**